### DISCUSSION TITLE
[Yocto] nerdctl run fails with pivot_root

### DISCUSSION THREAD

**Discussion by manojvi** [containerd/nerdctl#3731] [Q&A] [answered]
Hi experts,
looking to figure out what I could be missing while running nerdctl in this custom yocto environment.
I am running yocto on an airgapped machine. I have a local private registry and able to run :
`docker run --rm localhost:5000/pause:3.9` works fine and the `containerd`  and `containerd-shim` seem to be running.

When I attempt to launch nerdctl via `nerdctl run --rm localhost:5000/pause:3.9`  I get the below error:
```
 failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error jailing process inside rootfs: pivot_root .: invalid argument: unknown 
 ```
 
 I have tried setting `no_pivot=true`  on the `/etc/containerd/config.toml` but its still trying to do a `pivot_root`
 
 Software versions
 ```
#nerdctl --version
nerdctl version 1.7.7
#containerd --version
containerd github.com/containerd/containerd v1.7.23 57f17b0a6295a39009d861b89e3b3b87b005ca27
# runc --version
runc version 1.1.13
commit: v1.1.13-0-g58aa9203-dirty
spec: 1.0.2-dev
go: go1.21.11
libseccomp: 2.5.5

 ```

**Comment by AkihiroSuda** [comment]
This may work?
- https://github.com/opencontainers/runc/pull/4434

**Comment by manojvi** [comment, marked as answer]
Thanks! I could not apply the patch as is, but it gave me hints about the yocto booting up on a ramfs. After flashing the device to boot from disk, the Pivot_root issue appears to be resolved. Thanks
